warehouse/forklift/legacy.py

```diff
@@ -676,6 +676,22 @@ def file_upload(request):
                 ),
             ) from None
 
+    # Verify any verifiable URLs
+    publisher_base_url = (
+        request.oidc_publisher.publisher_base_url if request.oidc_publisher else None
+    )
+    project_urls = (
+        {}
+        if not meta.project_urls
+        else {
+            name: {
+                "url": url,
+                "verified": publisher_base_url
+                and url.lower().startswith(publisher_base_url.lower()),
```

Comment:
There's a bug here (and in the previous version) because checking for a prefix isn't enough. @facutuesca will be updating this with changes from https://github.com/trail-of-forks/warehouse/pull/2833.

Reply:
Since I don't have permissions for the repo where this branch lives, I created a new branch here: https://github.com/pypi/warehouse/pull/16205